aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -563,6 +563,19 @@ Tensor scatter_add(const Tensor & self, int64_t dim, const Tensor & index, const
   return self.clone(at::MemoryFormat::Preserve).scatter_add_(dim, index, source);
 }
 
+Tensor scatter_add_src_backward(
+  const Tensor& grad, int64_t dim, const Tensor& index,
+  IntArrayRef src_sizes, bool sparse_grad
+) {
+  auto grad_src = at::zeros(src_sizes, grad.options()).contiguous();
+  auto grad_contiguous = grad.contiguous();
+
+  gather_stub(grad_src.device().type(), grad_src, grad_contiguous, dim, index);
```

Comment:
Is the idea here that index will be expanded to the right sizes inside this call?

Reply:
No, index should not change, it is the output of gather that should be expanded.

Reply:
Yes but because the output you defined here is bigger than the indices, then the indices will be broadcasted?

Reply:
Here goes the expression:
https://github.com/pytorch/pytorch/issues/27614#issuecomment-564648819

Reply:
No, index should not broadcast.